### DISCUSSION TITLE
When the number of my files reaches 2.5 million,the writing is abnormally slow

### DISCUSSION THREAD

**Discussion by ponycloud235** [minio/minio#13798] [Q&A] [answered]
When the number of my files reaches 2.5 million and each file size is about 50K, the writing is abnormally slow.
![image](https://user-images.githubusercontent.com/2733701/144194417-a6e18d64-3201-4aac-9890-acca8b855bd5.png)

Files using mc client 1MB take about 21 seconds
![image](https://user-images.githubusercontent.com/2733701/144192813-b17fe997-1a67-4d34-b65b-662f5944892c.png)
Writing using java SDK is basically more than 10s
Storage is mounted using NFS，The network is normal
![image](https://user-images.githubusercontent.com/2733701/144194232-0630b4c2-8552-4c4e-b2e0-f7ac2da084e1.png)

now，How do I find the problem？

* minio version RELEASE.2021-11-05T09-16-26Z
* Linux ZNJC-WEB-02 3.10.0-1127.el7.x86_64 #1 SMP Tue Mar 31 23:36:51 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

**Comment by harshavardhana** [comment]
> * minio version RELEASE.2021-11-05T09-16-26Z
> * Linux ZNJC-WEB-02 3.10.0-1127.el7.x86_64 [Full restructure in accordance with #1](https://github.com/minio/minio/pull/1) SMP Tue Mar 31 23:36:51 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

Did you write all objects at the same prefix? with no hierarchy? @yihuiluo235

**Reply by ponycloud235** [reply]
No, there are different prefixes, by business, time and equipment

**Comment by harshavardhana** [comment, marked as answer]
> Storage is mounted using NFS，The network is normal

Using NFS for anything is a bad idea,  it is your NFS that is slow.

**Reply by ponycloud235** [reply]
It is normal to operate directly on nfs

**Reply by ponycloud235** [reply]
Is there a detailed log to query the problem? check log, only error information

**Reply by klauspost** [reply]
Ask NFS authors. MinIO doesn't do anything magical, so the FS determines your speed.

**Reply by ponycloud235** [reply]
thank you ,I try later
